x/ccv/provider/keeper/keeper.go

```diff
@@ -1184,3 +1184,141 @@ func (k Keeper) IsOptIn(ctx sdk.Context, chainID string) bool {
 	topN, found := k.GetTopN(ctx, chainID)
 	return !found || topN == 0
```

Comment:
Is it possible to have a chain with TopN set to `0`?

Reply:
I think it exactly indicates opt in chains.
Probably, this function should error on !found, and otherwise return topN == 0

Reply:
> Probably, this function should error on !found, and otherwise return topN == 0
Might augment this function to return an `error` later on.